src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/PresentationBuildTasks.csproj

```diff
@@ -308,14 +298,16 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkPackageVersion)" />
-    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCorePackageVersion)" />
-    <PackageReference Include="System.Reflection.MetadataLoadContext" Version="$(SystemReflectionMetadataLoadContextVersion)" />
+    <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkPackageVersion)" IncludeAssets="compile" />
+    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCorePackageVersion)" IncludeAssets="compile" />
 
-    <!--
-      Provide specific/old versions for PresentationBuildTasks which needs to run inside MSBuild
-    -->
-    <PackageReference Include="System.CodeDom" Version="$(SystemCodeDomPackageVersionForPresentationBuildTasks)" />
+    <!-- Reference (don't redistribute) the live version of SRM on .NETCoreApp and fallback to the toolset version for .NET Framework. -->
+    <PackageReference Include="System.Reflection.MetadataLoadContext" IncludeAssets="compile">
+      <Version Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'">$(SystemReflectionMetadataLoadContextVersion)</Version>
```

Comment:
You could use the toolset version in both cases.   I suspect it needs to be available for source build regardless.

Reply:
I'm following the dotnet/sdk pattern which uses live versions by default (via CPM) and downgrade only when targeting .NET Framework. As this repo doesn't use CPM, the pattern here looks differently.

> I suspect it needs to be available for source build regardless.

The windows UI stack isn't part of source-build.

Reply:
Sorry, I was just trying to guess at why you might have wanted to reference the latest on core.

The SDK actually needs to distribute live versions itself, while tasks could stay on older versions.